histogram/src/snapshot.rs

```diff
@@ -0,0 +1,84 @@
+//! A snapshot captures the distribution across a time range.
+
+use crate::{Bucket, Error, Histogram, Range, UnixInstant};
+
+/// An immutable snapshot of a distribution for a fixed time window.
+pub struct Snapshot {
+    pub(crate) range: Range<UnixInstant>,
+    pub(crate) histogram: Histogram,
+}
+
+impl Snapshot {
+    /// Get a reference to the raw counters.
+    pub fn as_slice(&self) -> &[u64] {
+        &self.histogram.buckets
+    }
+
+    /// Return a collection of percentiles from this histogram.
+    ///
+    /// Each percentile should be in the inclusive range `0.0..=100.0`. For
+    /// example, the 50th percentile (median) can be found using `50.0`.
+    ///
+    /// The results will be sorted by the percentile.
+    pub fn percentiles(&self, percentiles: &[f64]) -> Result<Vec<(f64, Bucket)>, Error> {
+        self.histogram.percentiles(percentiles)
+    }
+
+    /// Return a single percentile from this histogram.
+    ///
+    /// The percentile should be in the inclusive range `0.0..=100.0`. For
+    /// example, the 50th percentile (median) can be found using `50.0`.
+    pub fn percentile(&self, percentile: f64) -> Result<Bucket, Error> {
+        self.percentiles(&[percentile])
```

Comment:
Wouldn't `self.histogram.percentile` be easier?

Reply:
Yes, yes it would.